items:
- uid: System.IO.EndOfStreamException
  commentId: T:System.IO.EndOfStreamException
  id: EndOfStreamException
  parent: System.IO
  children:
  - System.IO.EndOfStreamException.#ctor
  - System.IO.EndOfStreamException.#ctor(System.String)
  - System.IO.EndOfStreamException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  - vb
  name: EndOfStreamException
  fullName: System.IO.EndOfStreamException
  type: Class
  source:
    remote:
      path: src/System.IO/ref/System.IO.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: EndOfStreamException
    path: src/System.IO/ref/System.IO.cs
    startLine: 103
  assemblies:
  - System.IO
  namespace: System.IO
  summary: "\n<p>The exception that is thrown when reading is attempted past the end of a stream.</p>\n"
  remarks: "\n<p>\n<strong>EndOfStreamException</strong> uses the HRESULT COR_E_ENDOFSTREAM which has the value 0x80070026.</p>\n"
  example:
  - "\n<p>The following code example shows how to read and write <strong>Double</strong> data to memory by using the <xref href=\"System.IO.BinaryReader\" data-throw-if-not-resolved=\"false\"></xref> and <xref href=\"System.IO.BinaryWriter\" data-throw-if-not-resolved=\"false\"></xref> classes on top of the <xref href=\"System.IO.MemoryStream\" data-throw-if-not-resolved=\"false\"></xref> class.</p>System.IO.BinaryReaderWriter.RWDouble#1"
  syntax:
    content: 'public class EndOfStreamException : IOException'
    content.vb: >-
      Public Class EndOfStreamException
          Inherits IOException
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  inheritance:
  - System.Object
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
- uid: System.IO.EndOfStreamException.#ctor
  commentId: M:System.IO.EndOfStreamException.#ctor
  id: '#ctor'
  parent: System.IO.EndOfStreamException
  langs:
  - csharp
  - vb
  name: EndOfStreamException()
  fullName: System.IO.EndOfStreamException.EndOfStreamException()
  type: Constructor
  source:
    remote:
      path: src/System.IO/ref/System.IO.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.IO/ref/System.IO.cs
    startLine: 105
  assemblies:
  - System.IO
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.EndOfStreamException\" data-throw-if-not-resolved=\"false\"></xref> class with its message string set to a system-supplied message and its HRESULT set to COR_E_ENDOFSTREAM.</p>\n"
  syntax:
    content: public EndOfStreamException()
    content.vb: Public Sub New
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.IO.EndOfStreamException.#ctor(System.String)
  commentId: M:System.IO.EndOfStreamException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.IO.EndOfStreamException
  langs:
  - csharp
  - vb
  name: EndOfStreamException(String)
  fullName: System.IO.EndOfStreamException.EndOfStreamException(System.String)
  type: Constructor
  source:
    remote:
      path: src/System.IO/ref/System.IO.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.IO/ref/System.IO.cs
    startLine: 106
  assemblies:
  - System.IO
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.EndOfStreamException\" data-throw-if-not-resolved=\"false\"></xref> class with its message string set to <em>message</em> and its HRESULT set to COR_E_ENDOFSTREAM.</p>\n"
  syntax:
    content: public EndOfStreamException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>A string that describes the error. The content of <em>message</em> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.IO.EndOfStreamException.#ctor(System.String,System.Exception)
  commentId: M:System.IO.EndOfStreamException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.IO.EndOfStreamException
  langs:
  - csharp
  - vb
  name: EndOfStreamException(String, Exception)
  fullName: System.IO.EndOfStreamException.EndOfStreamException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/System.IO/ref/System.IO.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.IO/ref/System.IO.cs
    startLine: 107
  assemblies:
  - System.IO
  namespace: System.IO
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.IO.EndOfStreamException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  remarks: "\n<p>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property. The <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property returns the same value that is passed into the constructor, or <strong>null</strong> if the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property does not supply the inner exception value to the constructor.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"System.IO.EndOfStreamException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The inner exception reference. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public EndOfStreamException(string message, Exception innerException)
    content.vb: Public Sub New(message As String, innerException As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>A string that describes the error. The content of <em>message</em> is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture. </p>\n"
    - id: innerException
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If the <em>innerException</em> parameter is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception. </p>\n"
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
references:
- uid: System.IO
  isExternal: false
  name: System.IO
  fullName: System.IO
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
